src/main/java/io/strimzi/kafka/bridge/tracing/TracingHandle.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package io.strimzi.kafka.bridge.tracing;
+
+import io.strimzi.kafka.bridge.config.BridgeConfig;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.kafka.client.consumer.KafkaConsumerRecord;
+
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * Simple interface to abstract tracing between legacy OpenTracing and new OpenTelemetry.
+ */
+public interface TracingHandle {
+    /**
+     * Tracing env var service name.
+     *
+     * @return tracing env var service name
+     */
+    String envServiceName();
+
+    /**
+     * Extract service name from bridge confing.
+     *
+     * @param config the bridge config
+     * @return bridge's service name
+     */
+    String serviceName(BridgeConfig config);
+
+    /**
+     * Initialize tracing.
+     */
+    void initialize();
+
+    /**
+     * Adapt executor service if needed.
+     * Else return null.
+     *
+     * @param service current executor service
+     * @return adapted executor service or null
+     */
+    default ExecutorService adapt(ExecutorService service) {
+        return null;
```

Comment:
not sure the logic is right. If not adapting it should return the executor service as it is. 
It would also mean simplifying the logic in the delagation we have in the  `BridgeExecutorServiceFactory` ...

```java
private ExecutorService delegate() {
     return TracingUtil.getTracing().adapt(original);
}
```

Reply:
Ah yes, that equally works, let me simplify it.